### DISCUSSION TITLE
sideload locales

### DISCUSSION THREAD

**Discussion by yoss770** [requarks/wiki#5306] [Error / Bug Report]
hi there

i'm using your app in kubernetes cluster using sideload with mountVolume .
the files located in pod under  /wiki/data/sideload witn the master as you ask in the documents.
unforetnely the web app not let me to choose another lang and not recognize the files in the folder .

![image](https://user-images.githubusercontent.com/54832691/168861325-55076265-5831-4f71-bfe3-b758aed52ccb.png)

![image](https://user-images.githubusercontent.com/54832691/168861774-8ea60b0c-20c5-4080-9ebd-18d1978317e0.png)

what can i do ??

thx to helpers

**Comment by yoss770** [comment]
just to make it clear i have reset the pod a few times !

**Comment by NGPixel** [comment]
The files should be directly under `/wiki/data/sideload`, not under an extra "wiki-locales" directory.

**Reply by yoss770** [reply]
hi NGPixel 

first of all big thank you !!!
i did  and restart the pod and nothing have change

![image](https://user-images.githubusercontent.com/54832691/169151244-56941a70-563b-4b38-8fb8-5cbc6a4094bd.png)

![image](https://user-images.githubusercontent.com/54832691/169151371-b86d3e6e-bfeb-4ed9-8b16-8603f73a91c8.png)


please let me know if have any idea for me please.

again THX.

**Reply by NGPixel** [reply]
Did you set `offline: true` in your config.yml?

**Comment by yoss770** [comment]
hi 
first off all big tnx for your fast replay.
do yo mean to this part in values.yaml :

{# This will allow us to install locales even without internet access using a initContainer & wikjs "sideloading"
**sideload:
  enabled: true**
  {# Git-Repo containing all locales.json-files you need:
  {#repoURL: https://github.com/Requarks/wiki-localization}

i did it before.

i can't find a value named   "offline: true"  

if not please let me know how to get this value please.

tnx a lot.

**Reply by rusLukasRath** [reply]
@yoss770 @NGPixel The problem is that the helm chart version does not support offline-mode/sideloading at all.

- When activating sideloading and defining my own repoURL in the `values.yaml`, the initContainer clones the repository, but the content is not shared to the actual running wiki.js container. => No locale files
- Regardless of the problem above, offline-mode is not set to true when activating sideloading from a different git repo.
- When mounting the locale files manually with an ConfigMap, offline-mode is not set to true in the config file. There is no way to change the config without having to mount a whole new file manually that overwrites the config.

Please see #4326 - being able to set offline-mode in the config is already fixed in that PR since August 2021

**Reply by yoss770** [reply]
hi rusLukasRath big tnx for your replay !

In your case, were you able to correct the languages ​​according to the changes in the code in the link you sent me?

I made all the changes and I get errors on the YAML.

 Error: UPGRADE FAILED: error validating "": error validating data: [ValidationError(Deployment.spec.template): unknown field "containers" in io.k8s.api.core.v1.PodTemplateSpec, ValidationError(Deployment.spec.template): unknown field "securityContext" in io.k8s.api.core.v1.PodTemplateSpec, ValidationError(Deployment.spec.template): unknown field "serviceAccountName" in io.k8s.api.core.v1.PodTemplateSpec, ValidationError(Deployment.spec.template): unknown field "volumes" in io.k8s.api.core.v1.PodTemplateSpec]

DEPLOYMENT.YAML

apiVersion: apps/v1
kind: Deployment
metadata:
  name: {{ include "wiki.fullname" . }}
  labels:
    {{- include "wiki.labels" . | nindent 4 }}
spec:
  replicas: {{ .Values.replicaCount }}
  selector:
    matchLabels:
      {{- include "wiki.selectorLabels" . | nindent 6 }}
  template:
    metadata:
      labels:
        {{- include "wiki.selectorLabels" . | nindent 8 }}
    spec:
    {{- with .Values.imagePullSecrets }}
      imagePullSecrets:
        {{- toYaml . | nindent 8 }}
    {{- end }}
      serviceAccountName: {{ include "wiki.serviceAccountName" . }}
      securityContext:
        {{- toYaml .Values.podSecurityContext | nindent 8 }}
      containers:
        - name: {{ .Chart.Name }}
          securityContext:
            {{- toYaml .Values.securityContext | nindent 12 }}
          image: "{{ .Values.image.repository }}:{{ default "latest" .Values.image.tag }}"
          imagePullPolicy: {{ default "IfNotPresent" .Values.image.imagePullPolicy }}
          env:
            - name: DB_TYPE
              value: postgres
            - name: DB_HOST
              value: {{ template "wiki.postgresql.host" . }}
            - name: DB_PORT
              value: "{{ default "5432" .Values.postgresql.postgresqlPort }}"
            - name: DB_NAME
              value: {{ default "wiki" .Values.postgresql.postgresqlDatabase }}
            - name: DB_USER
              value: {{ default "wiki" .Values.postgresql.postgresqlUser }}
            - name: DB_SSL
              value: "{{ default "false" .Values.postgresql.ssl }}"
            - name: DB_SSL_CA
              value: "{{ default "" .Values.postgresql.ca }}"
            - name: DB_PASS
              valueFrom:
                secretKeyRef:
                {{- if .Values.postgresql.existingSecret }}
                  name: {{ .Values.postgresql.existingSecret }}
                {{- else }}
                  name: {{ template "wiki.postgresql.secret" . }}
                {{- end }}
                  key: {{ template "wiki.postgresql.secretKey" . }}
            - name: OFFLINE_ACTIVE
              value: "{{ default "false" .Values.offlineMode }}"
          ports:
            - name: http
              containerPort: 3000
              protocol: TCP
          livenessProbe:
            {{- toYaml .Values.livenessProbe | nindent 12 }}
          readinessProbe:
            {{- toYaml .Values.readinessProbe | nindent 12 }}
          resources:
            {{- toYaml .Values.resources | nindent 12 }}
          volumeMounts:
            - name: config
              mountPath: /wiki/config.yml
              subPath: config.yml
              readOnly: true
      {{- with .Values.nodeSelector }}
      nodeSelector:
        {{- toYaml . | nindent 8 }}
      {{- end }}
    {{- with .Values.affinity }}
      affinity:
        {{- toYaml . | nindent 8 }}
    {{- end }}
    {{- with .Values.tolerations }}
      tolerations:
        {{- toYaml . | nindent 8 }}
    {{- end }}
      volumes:
      - name: config
        configMap:
          name: {{ include "wiki.fullname" . }}-config

i have create the new configmap file as in the link/
i have add the offlineMode value to values.yaml


i will be very happy to understand why its not working .

tnx again.

**Reply by rusLukasRath** [reply]
oh well, I was mentioning the PR for @NGPixel to take a look at it again because it would solve the config.yml problem when being implemented correctly into the current state. In your case, I would just create two ConfigMaps. One overwriting the config.yml with the correct settings and the other to inject the locale files into the sideloading folder.

**Comment by yoss770** [comment]
thanks Rohde !!

can you explain how?

i'm cant overwriting the config file in pod because is a "read only" file.

and the chart i attached before not working.

thx

**Reply by rusLukasRath** [reply]
Not really sure where you are stuck at. I can show you my kubernetes files that work for me.

ConfigMap with the locale files: https://pastebin.com/6yDpiKbP
ConfigMap with the config.yml that should be overwritten: https://pastebin.com/2GcwRmL7

and I have this in my `values.yaml`:
```
volumes:
  - name: wikijs-locale-config
    configMap:
      name: wikijs-locale-config
      defaultMode: 0777
  - name: wikijs-configyml-config
    configMap:
      name: wikijs-configyml-config
      defaultMode: 0777

volumeMounts:
  - name: wikijs-locale-config
    mountPath: /wiki/data/sideload
  - name: wikijs-configyml-config
    mountPath: /wiki/config.yml
    subPath: config.yml
```

**Reply by yoss770** [reply]
thanks Rohde !! 
just let me understand your way :  the configMap you have created will be in template  chart folder?  will be 2 files in same name?

thx

**Reply by therus000** [reply]
> Not really sure where you are stuck at. I can show you my kubernetes files that work for me.
> 
> ConfigMap with the locale files: https://pastebin.com/6yDpiKbP ConfigMap with the config.yml that should be overwritten: https://pastebin.com/2GcwRmL7
> 
> and I have this in my `values.yaml`:
> 
> ```
> volumes:
>   - name: wikijs-locale-config
>     configMap:
>       name: wikijs-locale-config
>       defaultMode: 0777
>   - name: wikijs-configyml-config
>     configMap:
>       name: wikijs-configyml-config
>       defaultMode: 0777
> 
> volumeMounts:
>   - name: wikijs-locale-config
>     mountPath: /wiki/data/sideload
>   - name: wikijs-configyml-config
>     mountPath: /wiki/config.yml
>     subPath: config.yml
> ```

in what folder you put those two files?

**Reply by therus000** [reply]
> > Not really sure where you are stuck at. I can show you my kubernetes files that work for me.
> > ConfigMap with the locale files: https://pastebin.com/6yDpiKbP ConfigMap with the config.yml that should be overwritten: https://pastebin.com/2GcwRmL7
> > and I have this in my `values.yaml`:
> > ```
> > volumes:
> >   - name: wikijs-locale-config
> >     configMap:
> >       name: wikijs-locale-config
> >       defaultMode: 0777
> >   - name: wikijs-configyml-config
> >     configMap:
> >       name: wikijs-configyml-config
> >       defaultMode: 0777
> > 
> > volumeMounts:
> >   - name: wikijs-locale-config
> >     mountPath: /wiki/data/sideload
> >   - name: wikijs-configyml-config
> >     mountPath: /wiki/config.yml
> >     subPath: config.yml
> > ```
> 
> in what folder you put those two files?

Error: helm upgrade have failed: parse error at (wiki/templates/wikijs-locale-config.yaml:34): function "bold" not defined

**Comment by divyabollampalli** [comment]
Hi i'm seeing this error 

024-01-29T20:19:45.710Z [MASTER] error: Syncing locales with Graph endpoint: [ FAILED ]
2024-01-29T20:19:45.711Z [MASTER] error: fetch failed
2024-01-29T20:19:45.712Z [MASTER] error: Fetching latest updates from Graph endpoint: [ FAILED ]
2024-01-29T20:19:45.712Z [MASTER] error: fetch failed

and the setup page looks like this 
<img width="1018" alt="Screenshot 2024-01-29 at 12 52 49 PM" src="https://github.com/requarks/wiki/assets/47512763/b2ab3e08-d17a-4283-878a-0aa3173109ce">

this is docker-compose file i'm using 
version: "3"
services:
  wikijs:
    image: wikijs_010924
    container_name: wikijs
    environment:
      - PUID=601
      - PGID=962
      - TZ=America/Los_Angeles
    volumes:
      - wikijs/sideload:/wiki/data/sideload
      - wikijs/config.yml:/wiki/config.yml
    ports:
      - 3000:3000
    restart: unless-stopped

please help me how to resolve the issue

**Reply by NGPixel** [reply]
Did you set `offline: true` in your config.yml?

**Reply by divyabollampalli** [reply]
yes i have set the offline:true


 Offline Mode
 ---------------------------------------------------------------------
 If your server cannot access the internet. Set to true and manually
download the offline files for sideloading.

offline: true

**Reply by divyabollampalli** [reply]
please tell me how to get it resolved

**Reply by NGPixel** [reply]
Have you tried restarting the container? and clearing your browser cache?

**Reply by divyabollampalli** [reply]
yes
![Uploading Screenshot 2024-01-29 at 2.29.46 PM.png…]()
 i did but still seeing the same issue

**Reply by divyabollampalli** [reply]
can you please help me understand the issue here  @NGPixel @rusLukasRath

**Comment by liviu-vasut** [comment]
I have created a PR to fix the sideload in the helm chart. Please review - https://github.com/requarks/wiki/pull/7963
